cmd/network-slirp-binding/domain/domain_test.go

```diff
@@ -141,28 +141,73 @@ var _ = Describe("QEMU slirp networking", func() {
+				vmschema.Interface{Name: "slirpTest", InterfaceBindingMethod: vmschema.InterfaceBindingMethod{Slirp: &vmschema.InterfaceSlirp{}}},
+				vmschema.Network{Name: "slirpTest", NetworkSource: vmschema.NetworkSource{Pod: &vmschema.PodNetwork{}}},
+				[]domainschema.Arg{
+					{Value: "-device"}, {Value: "foo"},
+					{Value: "-M"},
+				},
+				[]domainschema.Arg{
+					{Value: "-device"}, {Value: "foo"},
+					{Value: "-M"},
+					{Value: `-netdev`}, {Value: `user,id=slirpTest,net=10.0.2.0/24,dnssearch=dns.com`},
+					{Value: `-device`}, {Value: `{"driver":"e1000","netdev":"slirpTest","id":"slirpTest"}`},
+				},
+			),
+			Entry("slirp iface qemu cmd args already exist",
+				vmschema.Interface{Name: "slirpTest", InterfaceBindingMethod: vmschema.InterfaceBindingMethod{Slirp: &vmschema.InterfaceSlirp{}}},
+				vmschema.Network{Name: "slirpTest", NetworkSource: vmschema.NetworkSource{Pod: &vmschema.PodNetwork{}}},
+				[]domainschema.Arg{
+					{Value: `-netdev`}, {Value: `user,id=slirpTest,net=10.0.2.0/24,dnssearch=dns.com`},
+					{Value: `-device`}, {Value: `{"driver":"e1000","netdev":"slirpTest","id":"slirpTest"}`},
+				},
+				[]domainschema.Arg{
+					{Value: `-netdev`}, {Value: `user,id=slirpTest,net=10.0.2.0/24,dnssearch=dns.com`},
+					{Value: `-device`}, {Value: `{"driver":"e1000","netdev":"slirpTest","id":"slirpTest"}`},
+				},
+			),
+		)
+
+		It("should set QEMU cmd args correctly when executed more than once", func() {
 			iface := vmschema.Interface{Name: "slirpTest", InterfaceBindingMethod: vmschema.InterfaceBindingMethod{Slirp: &vmschema.InterfaceSlirp{}}}
+			network := vmschema.Network{Name: "slirpTest", NetworkSource: vmschema.NetworkSource{Pod: &vmschema.PodNetwork{}}}
+			existingArgs := []domainschema.Arg{
+				{Value: "-device"}, {Value: "foo"},
+				{Value: "-M"},
+			}
 
 			testMutator, err := domain.NewSlirpNetworkConfigurator([]vmschema.Interface{iface}, []vmschema.Network{network}, testSearchDomain)
 			Expect(err).ToNot(HaveOccurred())
 
-			expectedArg := "abc"
 			domSpec := &domainschema.DomainSpec{
-				QEMUCmd: &domainschema.Commandline{QEMUArg: []domainschema.Arg{{Value: expectedArg}}},
+				QEMUCmd: &domainschema.Commandline{QEMUArg: existingArgs},
 			}
 
-			mutatedDomSpec, err := testMutator.Mutate(domSpec)
-			Expect(err).ToNot(HaveOccurred())
-
-			Expect(mutatedDomSpec.QEMUCmd.QEMUArg).To(Equal(
-				[]domainschema.Arg{
-					{Value: expectedArg},
-					{Value: `-netdev`}, {Value: `user,id=slirpTest,net=10.0.2.0/24,dnssearch=dns.com`},
-					{Value: `-device`}, {Value: `{"driver":"e1000","netdev":"slirpTest","id":"slirpTest"}`},
-				},
-			))
+			expectedArgs := append(
+				existingArgs,
+				domainschema.Arg{Value: `-netdev`}, domainschema.Arg{Value: `user,id=slirpTest,net=10.0.2.0/24,dnssearch=dns.com`},
+				domainschema.Arg{Value: `-device`}, domainschema.Arg{Value: `{"driver":"e1000","netdev":"slirpTest","id":"slirpTest"}`},
+			)
+			for i := 0; i < 5; i++ {
```

Comment:
I think it is enough to execute this twice without a loop. The loop raises questions like: Why a loop was needed, what is special about 5?

Reply:
Why not using a loop? even for executing it twice?

I picked 5 arbitrarily since its cost nothing.

Reply:
Because a loop causes the reader to think there is more to it.
5 does the same, it causes the reader to think there is something special about this number.

We prefer to see no loops in tests, they hint of some kind of logic, which we usually do not want to have in tests. Loops sometimes are a must, but in this case I do not think we need them. In any case, repeating twice should be enough.

Reply:
This test is all about performing an operation repeatedly using loops should be expected.
Although, I don't care much about it, fixed.